packages/components/src/2016-12-01/button.js

```diff
@@ -27,13 +28,75 @@ class Button extends HTMLElement {
     shadowRoot.innerHTML = `
       <orion-inline
         border-radius="2"
-        background="black"
-        color="white"
+        background="white"
+        color="black"
         padding-horizontal="3"
         padding-vertical="2"
-        dim
         pointer><slot /></orion-inline>
     `;
+    this.shadowEl = shadowRoot.children[0];
+
+    this.addEventListener('mouseenter', () => {
+      if (this.state.disabled) { return; }
+      const nextState = ButtonState.enterHover(this.state);
+      this.dispatchEvent(new CustomEvent('change', { detail: nextState }));
+    });
+
+    this.addEventListener('mouseleave', () => {
+      const nextState = ButtonState.leaveHover(this.state);
+      this.dispatchEvent(new CustomEvent('change', { detail: nextState }));
+    });
+
+    this.addEventListener('click', (event) => {
+      if (this.state.disabled) {
+        event.stopImmediatePropagation();
+      }
+    });
+  }
+
+  get state() {
+    return {
+      hover: this.getAttribute('hover') === 'true',
+      disabled: this.getAttribute('disabled') === 'true',
+    };
+  }
+
+  static get observedAttributes() {
+    return ['background', 'color', 'disabled', 'hover'];
```

Comment:
Is the intent with this encapsulated state to expose as properties all state values, including ones that would otherwise be considered private? Case in point, `hover` would normally be private, but now we're exposing it through an observable attribute on the custom element. This doesn't seem like a big deal with a simple element like a button, but it could get quite gnarly with more complex ones.